state/src/item/asset_scheme.rs

```diff
@@ -53,33 +56,43 @@ impl AssetScheme {
         self.registrar.is_some()
     }
 
-    pub fn init(&mut self, metadata: String, amount: u64, registrar: Option<Address>) {
+    pub fn init(&mut self, metadata: String, amount: u64, registrar: Option<Address>, pool: Vec<Asset>) {
         assert_eq!("", &self.metadata);
         assert_eq!(0, self.amount);
         assert_eq!(None, self.registrar);
         self.metadata = metadata;
         self.amount = amount;
         self.registrar = registrar;
+        self.pool = pool;
+    }
+
+    pub fn pool(&self) -> &Vec<Asset> {
```

Comment:
```rust
pub fn pool(&self) -> &[Asset] {
```

Reply:
I changed it.

Reply:
It seems `&[Asset]` and `&Vec<Asset>` is the same type. Could you explain why `&[Asset]` is better?

Reply:
Slice and Vector are not the same type. They are different as String vs str.
https://www.reddit.com/r/rust/comments/8kujd1/newbie_correct_way_to_pass_vector_of_references/